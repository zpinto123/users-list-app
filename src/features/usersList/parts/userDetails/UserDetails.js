import React from "react";
import PropTypes from "prop-types";

import avatar from "../../../../assets/images/avatar_user.png";

import "./UserDetails.css";

const UserDetails = ({ userInfo: { name, surname, email, employer } }) => (
  <div className="userDetailsContainer">
    <div className="userAvatarContainer">
      <img className="userAvatar" src={avatar} alt={name} />
    </div>
    <div className="userInfoContainer">
      <div className="userInfo">
        <label className="label">Name: </label>
        <label className="value">{name || "--"}</label>
      </div>
      <div className="userInfo">
        <label className="label">Surname: </label>
        <label className="value">{surname || "--"}</label>
      </div>
      <div className="userInfo">
        <label className="label">Email: </label>
        <label className="value">{email || "--"}</label>
      </div>
      <div className="userInfo">
        <label className="label">Employer: </label>
        <label className="value">{employer || "--"}</label>
      </div>
    </div>
  </div>
);

UserDetails.propTypes = {
  userInfo: PropTypes.shape({
    name: PropTypes.string.isRequired,
    surname: PropTypes.string.isRequired,
    email: PropTypes.string,
    employer: PropTypes.string
  }),
};

export default UserDetails;
