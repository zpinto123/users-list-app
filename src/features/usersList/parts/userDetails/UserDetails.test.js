import React from "react";
import renderer from "react-test-renderer";

import UserDetails from "./UserDetails";
describe("<UserDetails />", () => {
  it("renders correctly", () => {
    const props = {
      userInfo: {
        displayName: "José Pinto",
        name: "José Pinto",
        surname: "Pinto",
        email: "super@super.com",
        employer: "JpanOTECH"
      }
    };

    const component = renderer.create(<UserDetails {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
