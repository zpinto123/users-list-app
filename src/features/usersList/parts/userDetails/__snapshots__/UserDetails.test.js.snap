// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetails /> renders correctly 1`] = `
<div
  className="userDetailsContainer"
>
  <div
    className="userAvatarContainer"
  >
    <img
      alt="José Pinto"
      className="userAvatar"
      src="avatar_user.png"
    />
  </div>
  <div
    className="userInfoContainer"
  >
    <div
      className="userInfo"
    >
      <label
        className="label"
      >
        Name: 
      </label>
      <label
        className="value"
      >
        José Pinto
      </label>
    </div>
    <div
      className="userInfo"
    >
      <label
        className="label"
      >
        Surname: 
      </label>
      <label
        className="value"
      >
        Pinto
      </label>
    </div>
    <div
      className="userInfo"
    >
      <label
        className="label"
      >
        Email: 
      </label>
      <label
        className="value"
      >
        super@super.com
      </label>
    </div>
    <div
      className="userInfo"
    >
      <label
        className="label"
      >
        Employer: 
      </label>
      <label
        className="value"
      >
        JpanOTECH
      </label>
    </div>
  </div>
</div>
`;
